library/x509_crt.c

```diff
@@ -1162,7 +1162,7 @@ int mbedtls_x509_crt_parse_path( mbedtls_x509_crt *chain, const char *path )
     int t_ret;
     struct stat sb;
     struct dirent *entry;
-    char entry_name[255];
+    char entry_name[MBEDTLS_X509_ENTRY_NAME_LEN];
     DIR *dir = opendir( path );
```

Comment:
I like the fix, but reading the function I think we should be checking if the `snprintf()` at line 1181 has been successful, and if not returning an error. Could you please add in the additional check to this PR?